lib/TTT/game.ex

```diff
@@ -5,32 +5,40 @@ defmodule Game do
 
   def start(io) do
     io.display(Messages.get_message(:welcome))
+    state = get_options(io)
+    play(state, io, state.mode)
+  end
+
+  def get_options(io) do
     mode = select_mode(Validation.choose_mode(io))
     player_token = Validation.choose_token(io)
-    opponent_symbol = Validation.choose_token(io, player_token)
-    state = set_options(player_token, opponent_symbol, mode)
-    play(state, io, mode)
+    opponent_token = Validation.choose_token(io, player_token)
+    current_player = Validation.choose_first_player(io, player_token, opponent_token)
+    set_options(player_token, opponent_token, current_player, mode)
   end
 
-  def set_options(player, opponent, mode) do
-    %State{player: player, opponent: opponent, current_player: player, mode: mode}
+  def set_options(player, opponent, current_player, mode) do
+    %State{player: player, opponent: opponent, current_player: current_player, mode: mode}
   end
 
   def select_mode(mode) when mode == "1", do: @mode1
   def select_mode(mode) when mode == "2", do: @mode2
   def select_mode(mode) when mode == "3", do: @mode3
 
-  def over(io, state) do
-    Messages.display_board(io, state)
-    Messages.end_of_game(io, state)
-  end
-
   def play(state, io, mode) do
     Messages.display_board(io, state)
+    player = state.player
+    opponent = state.opponent
 
-    human_move(state, io)
-    |> opponent(io, mode)
-    |> check_status(io, mode)
+    case state.current_player do
+      ^player ->
+        human_move(state, io, mode)
+        |> check_status(io, mode)
+
+      ^opponent ->
+        opponent(state, io, mode)
```

Comment:
This is pretty minor but I'd advise against overloading an overloaded function with a variable of the same name